pool.go

```diff
@@ -0,0 +1,74 @@
+package Go_Pool
+
+import (
+	"errors"
+	"runtime"
+	"sync"
+)
+
+type Pool struct {
+	concurrency      int
+	numCPUs          int
+	tasksChan        chan *Task
+	activeGoRoutines chan bool
+	wg               sync.WaitGroup
+}
+
+var pool *Pool
+var once sync.Once
+
+func NewPool(concurrency int, numCPUs int) *Pool {
+	once.Do(func() {
+		pool = &Pool{
+			concurrency:      concurrency,
+			tasksChan:        make(chan *Task, concurrency),
+			activeGoRoutines: make(chan bool, concurrency),
+		}
+	})
+	pool.SetNumCPUs(numCPUs)
+	go pool.schedule()
```

Comment:
Is this required every time the function is called?

Reply:
No required and changed it.